static/js/sent_messages.js

```diff
@@ -110,7 +110,7 @@ class MessageState {
         // wrong with our event loop.
 
         if (!this.received) {
-            setTimeout(this.maybe_restart_event_loop, 5000);
+            setTimeout(this.maybe_restart_event_loop.bind(this), 5000);
```

Comment:
@andersk can you review this one?  I'm not sure whether there's any audit we should do for similar issues.

(Or if there's a better refactor we should do to replace `this` with a better class structure).

Reply:
(Or @amanagr if you have ideas for a cleaner way to write this).

Reply:
AFAIK, this is the best method to do so. I learned it from https://javascript.info/bind.

Reply:
FTR, one alternative is `setTimeout(() => this.maybe_restart_event_loop(), 5000)`, but `.bind(this)` is also pretty idiomatic.